src/Angular.js

```diff
@@ -901,7 +901,7 @@ function copy(source, destination) {
       case '[object Uint8ClampedArray]':
       case '[object Uint16Array]':
       case '[object Uint32Array]':
-        return new source.constructor(copyElement(source.buffer));
+        return new source.constructor(copyElement(source.buffer), source.byteOffset, source.length);
```

Comment:
@gkalpak nope, `source.byteLength = source.length * source.BYTES_PER_ELEMENT`.
I've made some sample code to demo why `source.length` and not `source.byteLength`:
https://plnkr.co/edit/afWbk8OboJopB6BgSri7